tsconfig.json

```diff
@@ -1,11 +1,14 @@
 {
-    "compilerOptions": {
-        "module": "commonjs",
-        "target": "es2015",
-        "declaration": true,
-        "outDir": "./dist"
-    },
-    "include": [
-        "src/**/*"
-    ]
+  "compilerOptions": {
+    "module": "commonjs",
+    "target": "es5",
+    "declaration": true,
+    "outDir": "./lib",
+    "strict": true,
+    "jsx": "react",
+    "typeRoots": ["./types", "node_modules/@types"],
+    "esModuleInterop": true,
+    "keyofStringsOnly": true
```

Comment:
This is an interesting setting I read about in the O'Reilly Programming Typescript book (different from the Effective Typescript book)

This avoids those `string | number | Symbol` typing issues with object keys by saying "just treat object keys as strings" I guess (it accomplished being able to remove eg force-typing `propName as string` (where `propName: TPropName` and `TPropName extends keyof TProps`) in a couple places)